clang/lib/Driver/ToolChains/HIP.cpp

```diff
@@ -112,8 +112,10 @@ void AMDGCN::constructHIPFatbinCommand(Compilation &C, const JobAction &JA,
   // for backward compatibility. For code object version 4 and greater, the
   // offload kind in bundle ID is 'hipv4'.
   std::string OffloadKind = "hip";
+  // bundle ID equals 'hip' is always right.
   if (getAMDGPUCodeObjectVersion(C.getDriver(), Args) >= 4)
-    OffloadKind = OffloadKind + "v4";
+    // OffloadKind = OffloadKind + "v4";
+    OffloadKind = OffloadKind;
```

Comment:
It would be preferable if the no-op code can be replaced by an explanatory comment.

Reply:
> It would be preferable if the no-op code can be replaced by an explanatory comment.

More reasonable solution is:
`if (haveAMDGPUCodeObjectVersionArgument(C.getDriver(), Args) && getAMDGPUCodeObjectVersion(C.getDriver(), Args) >= 4)
    OffloadKind = OffloadKind + "v4";`

Reply:
> Could some minimal driver unit testing be added? I believe the following checks can be run:
> 
> 1. Phases printout upon receiving the AMDCGN triple
> 2. Checks for the correct library/linker paths & options, as set within `HIPToolChain::addClangTargetOptions()`
>    This could be done within a separate `clang/test/Driver/sycl-offload-amdcgn.cpp` file, or somehow unified with the NVPTX testing on the premises of CUDA/HIP compatibility between the targets (based on your vision and preference)

Do you mean add `CudaInstallationDetector` class like clang/lib/driver/cuda.cpp?